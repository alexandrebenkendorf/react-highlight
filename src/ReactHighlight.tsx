import Highlight, { HighlightConfig } from './Highlight';

type ReactHighlightProps = HighlightConfig & {
  searchTerm: string;
  text: string;
};
export default function ReactHighlight({
  searchTerm,
  text,
  caseSensitive,
  className,
  inlineStyle,
}: ReactHighlightProps) {
  const highlight = new Highlight({ caseSensitive, className, inlineStyle });
  const htmlString = highlight.term(searchTerm).in(text).html.innerHTML;
  return <span dangerouslySetInnerHTML={{ __html: htmlString }}></span>;
}
