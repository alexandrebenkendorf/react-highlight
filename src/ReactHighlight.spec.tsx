import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

import ReactHighlight from './ReactHighlight';

it('should render correctly', () => {
  const { container } = render(<ReactHighlight searchTerm="test" text="testing result" />);
  expect(container).toMatchSnapshot();
});
