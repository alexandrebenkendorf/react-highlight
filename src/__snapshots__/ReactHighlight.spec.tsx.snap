// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div>
  <span>
    <strong
      data-testid="highlighted"
    >
      test
    </strong>
    ing result
  </span>
</div>
`;
